jwst/master_background/master_background_step.py

```diff
@@ -73,7 +78,25 @@ def process(self, input):
                 self.record_step_status(result, 'master_background', success=False)
 
                 return result
+            # Check if subtract_background is set to False -> skip step
+            if self.subtract_background is False:
+                self.log.warning(
+                    "Not subtracting masterbackground, subtract_background set to False")
+                result = input_data.copy()
+                self.record_step_status(result, 'master_background', success=False)
+                return result
+            # Check if subtract_background is None but the background was
+            # subtracted in calspec2 background step  -> skip step
+            if self.subtract_background is None and \
+                input_data.meta.cal_step.back_sub == 'COMPLETE':
+                self.log.warning(
```

Comment:
I'm not sure if we want this to be a warning. The normal/default flow will likely be to have calspec2 step applied and hence this will be a fairly standard case (to have `master_background` get skipped). When users see a warning message they tend to immediately think that something's wrong.